executor/distsql.go

```diff
@@ -488,7 +478,11 @@ func (e *XSelectIndexExec) nextForSingleRead() (*Row, error) {
 			schema = e.idxColsSchema
 		}
 		values := make([]types.Datum, schema.Len())
-		err = codec.SetRawValues(rowData, values)
+		if e.handleCol != nil && e.handleCol.ID == model.ExtraHandleID {
```

Comment:
So a function is better? The function name could be used as the comment for this check.